execution/engine/engine.go

```diff
@@ -58,17 +59,23 @@ func New(
 	}
 }
 
-// Start spawns any goroutines required by the service.
-func (ee *Engine) Start(
-	ctx context.Context,
-) error {
+func (ee *Engine) Start(ctx context.Context) error {
+	errChan := make(chan error, 1)
 	go func() {
-		// TODO: handle better
-		if err := ee.ec.Start(ctx); err != nil {
-			panic(err)
-		}
+			if err := ee.ec.Start(ctx); err != nil {
+					ee.logger.Error("engine client failed to start", "error", err)
+					errChan <- err
+					return
+			}
+			close(errChan)
 	}()
-	return nil
+
+	select {
+	case err := <-errChan:
```

Comment:
This changes the behavior of `Start`. Previously, we'd exit as soon as goroutine was spawned. Now we're waiting for `Start` to finish (blocking select). What's the point of having a goroutine then?